lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,468 @@
+        #        increasing first-guess lightning probability to risk 2 (LR2).
+        self.lrt_lev2 = 0.
+
+        # Set values for handling precipitation rate data
+        #    precipthr (tuple):
+        #        Values for limiting prob(lightning) with prob(precip).
+        #        These are the three prob(precip) thresholds and are designed
+        #        to prevent a large probability of lightning being output if
+        #        the probability of precipitation is very low.
+        self.precipthr = (0.0, 0.05, 0.1)
+        #    ltngthr (tuple):
+        #        Values for limiting prob(lightning) with prob(precip)
+        #        These are the three prob(lightning) values to scale to.
+        self.ltngthr = (0.0067, 0.25, 1.)
+        #    probability thresholds for increasing the prob(lightning)
+        #        phighthresh for heavy precip (>7mm/hr)
+        #            relates to problightning_values[2]
+        #        ptorrthresh for intense precip (>35mm/hr)
+        #            relates to problightning_values[1]
+        self.phighthresh = 0.4
+        self.ptorrthresh = 0.2
+
+        # Set values for handling vertically-integrated-ice (VII) data
+        #    ice_thresholds (tuple):
+        #        Values for increasing prob(lightning) with column-ice data.
+        #        These are the three VII thresholds in kg/m2.
+        self.ice_thresholds = (0.5, 1.0, 2.0)
+        #    ice_scaling (tuple):
+        #        Values for increasing prob(lightning) with VII data.
+        #        These are the three prob(lightning) values to scale to.
+        self.ice_scaling = (0.1, 0.5, 0.9)
+
+    def __repr__(self):
+        """
+        Docstring to describe the repr, which should return a
+        printable representation of the object.
+        """
+        return """<NowcastLightning: radius={radius},
+ lightning mapping (lightning rate in "min^-1"):
+   upper: lightning rate {lthru} => min lightning prob {lprobu}
+   lower: lightning rate {lthrl} => min lightning prob {lprobl}
+>""".format(radius=self.radius,
+            lthru=self.lrt_lev1, lthrl=self.lrt_lev2,
+            lprobu=self.pl_dict[1], lprobl=self.pl_dict[2])
+
+    @staticmethod
+    def _update_metadata(cube):
+        """
+        Modify the meta data of input cube to resemble a Nowcast of lightning
+        probability.
+        1. Rename to "probability_of_lightning"
+        2. Remove "threshold" coord
+            (or causes iris.exceptions.CoordinateNotFoundError)
+        3. Discard all cell_methods
+
+        Args:
+            cube (iris.cube.Cube):
+                An input cube
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube - a copy of input cube with meta-data relating to
+                a Nowcast of lightning probability.
+                The data array will be a copy of the input cube.data
+        """
+        new_cube = cube.copy()
```

Comment:
This might be worth discussing with Bruce - we've been discussing / I sent him some notes on the different approaches to metadata we've implemented during this and last sprint.